src/connection.rs

```diff
@@ -11,104 +11,183 @@ use error::MemcacheError;
 use protocol::{AsciiProtocol, BinaryProtocol, Protocol};
 use stream::Stream;
 use stream::UdpStream;
+use openssl::ssl::{SslConnector, SslMethod, SslFiletype, SslVerifyMode};
 
 /// a connection to the memcached server
 pub struct Connection {
     pub protocol: Protocol,
     pub url: String,
 }
 
-impl Connection {
-    pub(crate) fn connect(url: &Url) -> Result<Self, MemcacheError> {
+enum Transport {
```

Comment:
Words like `TCP` / `UDP` / `TLS` should using lower case with first char upper case. Maybe all upper case is right, but for now we should keep these like other places in this project.